pkg/claimsstore/nodepersister.go

```diff
@@ -53,17 +54,26 @@ func (c *Node) ToKV() (db.KV, error) {
 }
 
 // UpdateKVAndPrefix takes a KV and updates the values on the node
-func (c *Node) UpdateKVAndPrefix(kv db.KV, prefix []byte) {
+func (c *Node) UpdateKVAndPrefix(kv db.KV, prefix []byte) error {
 	nodetType := merkletree.NodeType(kv.V[0])
 	var typeName string
 	if nodetType == merkletree.NodeTypeMiddle {
 		typeName = MiddleNode
 	} else if nodetType == merkletree.NodeTypeLeaf {
 		typeName = LeafNode
 	}
-	c.DID = string(prefix)
+	if bytes.Equal(prefix, PrefixRootMerkleTree) || len(prefix) != 32 {
```

Comment:
Is this only expecting our DID or can it handle other DIDs as well?

Reply:
It's only expecting our DID or UUID based DIDs, this part should maybe be rethought.  Iden3 has a ID class that is 31 bytes.  Maybe we use those and add that id to the did model. Or actually at this stage it should probably just hex encode whatever is there.  I thought it would be nice to be able to query this table by a did, but might make things too fragile.